### ISSUE TITLE
Provisioning not starting soon enough when done repeatedly 

### ISSUE THREAD

**Issue by Xenwar** [metal3-io/cluster-api-provider-metal3#42]
**What steps did you take and what happened:**
- Provision and de-Provision a control plane node few times. 

At some point, Provisioning does not start at all and the bmh states remains ready.

The waiting time passes 30 minutes. Normally, it should start within 10 minutes.
It is safe assume that the provisioning will not start at all.

**What did you expect to happen:**
- Each time, the provision and de-provision should start and succeed. 
- The process should be stateless in that previous provisioning should not have an impact on the current one for as long as there are free available bmh.

**Anything else you would like to add:**
- In order to fix the issue, one needs to run ```make clean && make```
- Once in a while, even that may not help. And, re-creating the metal3-dev-env machine is the only option
- the logs for relevant controllers are added as comment

**Environment:**

- Cluster-api version: v1alpha3
- CAPBM version: v0.3.0
- Minikube version: v1.8.2
- metal3-dev-env
- Kubernetes version: v1.17.3

/kind bug

**Comment by Xenwar** [comment]
capi-controller-manager-664c75c4df-lxczd.log
I0325 08:25:00.908046       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:25:00.909659       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:25:00.968667       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:25:00.969040       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:25:30.966094       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:25:30.967071       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:26:00.986737       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:26:00.987437       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:26:13.799765       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:26:13.800000       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:26:30.996818       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:26:30.997330       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:27:01.008122       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:27:01.008473       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:27:31.019744       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:27:31.020283       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:28:01.033265       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:28:01.033509       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:28:09.883675       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:28:09.883892       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:28:31.053705       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:28:31.054598       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:29:01.068057       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:29:01.068290       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:29:31.075527       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:29:31.075752       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:30:01.083024       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:30:01.083716       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:30:31.088593       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:30:31.088763       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:31:01.110806       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:31:01.111270       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:31:31.114522       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:31:31.114710       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:32:01.132329       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:32:01.133213       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:32:31.141069       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:32:31.141619       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:33:01.146628       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:33:01.147193       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:33:31.155430       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:33:31.155697       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:34:01.163277       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:34:01.163706       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
I0325 08:34:19.316447       1 machine_controller_noderef.go:53] controllers/Machine "msg"="Machine doesn't have a valid ProviderID yet" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3" 
E0325 08:34:19.316615       1 machine_controller.go:215] controllers/Machine "msg"="Reconciliation for Machine asked to requeue" "error"="Bootstrap provider for Machine \"test1-controlplane-dlxrc\" in namespace \"metal3\" is not ready, requeuing: requeue in 30s" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "namespace"="metal3"

**Comment by Xenwar** [comment]
```
capi-kubeadm-bootstrap-controller-manager-66597cb7f4-ghmqq.log
I0325 08:25:00.875004       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:25:00.875156       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61729" 
I0325 08:25:00.939434       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:25:00.940294       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61729" 
I0325 08:25:00.956331       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:25:00.956874       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61736" 
I0325 08:25:30.879739       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:25:30.880052       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:25:57.940518       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:25:57.940733       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:26:00.893857       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:26:00.894438       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:26:30.898614       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:26:30.899446       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:27:00.904902       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:27:00.905045       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:27:30.907157       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:27:30.908903       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:28:00.913552       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:28:00.913653       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:28:13.477557       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:28:13.477672       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:28:30.915758       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:28:30.915823       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:29:00.917180       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:29:00.917335       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:29:30.918432       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:29:30.918673       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:29:37.715265       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:29:37.723335       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:30:00.920169       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:30:00.920221       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:30:30.923608       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:30:30.925478       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:31:00.931909       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:31:00.932211       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:31:30.935369       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:31:30.935707       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:32:00.937187       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:32:00.937233       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:32:30.941261       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:32:30.941382       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:33:00.943393       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:33:00.943444       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:33:30.944568       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:33:30.945902       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 
I0325 08:34:00.947475       1 control_plane_init_mutex.go:74] init-locker "msg"="Waiting on another machine to initialize" "cluster-name"="test1" "configmap-name"="test1-lock" "machine-name"="test1-controlplane-dlxrc" "namespace"="metal3" "init-machine"="test1-controlplane-rsfz4"
I0325 08:34:00.947730       1 kubeadmconfig_controller.go:286] controllers/KubeadmConfig "msg"="A control plane is already being initialized, requeing until control plane is ready" "kind"="Machine" "kubeadmconfig"={"Namespace":"metal3","Name":"test1-controlplane-hlvtj"} "name"="test1-controlplane-dlxrc" "version"="61737" 

```

**Comment by Xenwar** [comment]
```
capi-kubeadm-control-plane-controller-manager-56d8897c7d-pp9kx.log

I0325 08:25:00.302863       1 controller.go:228] controllers/KubeadmControlPlane "msg"="Initializing control plane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" "Desired"=1 "Existing"=0
E0325 08:25:02.255521       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:02.303562       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:02.305022       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:08.464251       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:08.481711       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:08.483173       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:14.607820       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:14.630162       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:14.631498       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:20.757361       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:20.757850       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:20.760414       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:33.047321       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:33.048895       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:33.050910       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:39.184642       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:39.185304       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:39.185590       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:45.327885       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:45.329105       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:45.330829       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:58.576836       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:25:58.578458       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:25:58.578915       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:04.719572       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:04.720500       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:26:04.721810       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:10.863508       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:10.864383       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:26:10.865001       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:17.008907       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:17.011540       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:26:17.012763       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:23.152544       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:23.152832       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:26:23.153122       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:29.296110       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:29.296786       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:26:33.394245       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:38.704325       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:26:38.706822       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:27:19.667413       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:27:25.807789       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:27:25.811648       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:28:18.581730       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:28:24.368457       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:28:24.373475       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:28:47.732624       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:28:53.871504       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:28:53.873747       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}
I0325 08:29:42.868067       1 controller.go:170] controllers/KubeadmControlPlane "msg"="Reconcile KubeadmControlPlane" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:29:46.738665       1 controller.go:146] controllers/KubeadmControlPlane "msg"="Failed to update KubeadmControlPlane Status" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
E0325 08:29:46.739343       1 controller.go:258] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: no route to host"  "controller"="kubeadmcontrolplane" "request"={"Namespace":"metal3","Name":"test1-controlplane"}

```

**Comment by Xenwar** [comment]
```
capm3-controller-manager-5d6d8c75cd-pvgcz.log

I0325 08:25:00.337174       1 metal3machine_controller.go:98] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for Machine Controller to set OwnerRef on Metal3Machine" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:25:00.634846       1 metal3machine_controller.go:98] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for Machine Controller to set OwnerRef on Metal3Machine" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:25:00.899350       1 metal3machine_manager.go:135] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for the Bootstrap provider controller to set bootstrap data" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "metal3-cluster"="test1" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:25:00.918845       1 metal3machine_manager.go:135] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for the Bootstrap provider controller to set bootstrap data" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "metal3-cluster"="test1" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:25:00.936486       1 metal3machine_manager.go:135] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for the Bootstrap provider controller to set bootstrap data" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "metal3-cluster"="test1" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:25:00.953735       1 metal3machine_manager.go:135] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for the Bootstrap provider controller to set bootstrap data" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "metal3-cluster"="test1" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:26:09.730749       1 metal3machine_manager.go:135] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for the Bootstrap provider controller to set bootstrap data" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "metal3-cluster"="test1" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:27:39.599755       1 metal3cluster_controller.go:108] Metal3Cluster-controller "msg"="Reconciling metal3Cluster" "cluster"="test1" "metal3-cluster"={"Namespace":"metal3","Name":"test1"} 
I0325 08:27:39.613196       1 metal3cluster_controller.go:108] Metal3Cluster-controller "msg"="Reconciling metal3Cluster" "cluster"="test1" "metal3-cluster"={"Namespace":"metal3","Name":"test1"} 
I0325 08:28:25.107541       1 metal3machine_manager.go:135] controllers/Metal3Machine/Metal3Machine-controller "msg"="Waiting for the Bootstrap provider controller to set bootstrap data" "cluster"="test1" "machine"="test1-controlplane-dlxrc" "metal3-cluster"="test1" "metal3-machine"={"Namespace":"metal3","Name":"test1-controlplane-flzzw"} 
I0325 08:29:00.922826       1 metal3cluster_controller.go:108] Metal3Cluster-controller "msg"="Reconciling metal3Cluster" "cluster"="test1" "metal3-cluster"={"Namespace":"metal3","Name":"test1"} 
I0325 08:29:00.937756       1 metal3cluster_controller.go:108] Metal3Cluster-controller "msg"="Reconciling metal3Cluster" "cluster"="test1" "metal3-cluster"={"Namespace":"metal3","Name":"test1"} 

```

**Comment by maelk** [comment]
This is happening on master, or on the specified release ? how are you deploying ? because there is a bug that could cause this, but only with v1alpha4.

**Comment by Xenwar** [comment]
I am deploying as follows. 
```
git clone https://github.com/metal3-io/metal3-dev-env.git
cd metal3-dev-env
export CONTAINER_RUNTIME=docker
export DEFAULT_HOSTS_MEMORY=4096
export IMAGE_OS=Ubuntu
export CAPI_VERSION=v1alpha3
export NUM_NODES=4
make
```

**Comment by maelk** [comment]
ok, then this is something that needs to be investigated.

**Comment by maelk** [comment]
How many control plane nodes do you have ? If one only, you cannot deprovision and reprovision it without completely deleting the cluster. Otherwise it will try to use "kubeadm join" when the cluster has been deprovisioned. you can do that only if you have at least three masters. Is this your case ? and if it is, has the initial provisioning fully completed ?

**Comment by Xenwar** [comment]
I have only one control plane node provisioned and three others in ready state.
This use case is ```a single master upgrade with extra node(s)```. 
And, relevant to this issue is that fact that the work flow successful. as of 16.03.1020

If this is not a valid use case, then the task list needs to be revisited and use cases redefined again.

**Comment by Xenwar** [comment]
> ok, then this is something that needs to be investigated.

I agree. It might better to split this issue into two, if current investigations do not yield fruit. 
1. Why does it fail to create the new node successfully. 
2 .Why is the original node affected/changed before the success of the new one.

**Comment by stbenjam** [comment]
@Xenwar Any updates on this? Do you plan to split this issue out?

**Comment by Xenwar** [comment]
> @Xenwar Any updates on this? Do you plan to split this issue out?

Yes, I intend to split this issue into two. But, we have other issues blocking us from working on it. 
Will do that as soon as possible

**Comment by maelk** [comment]
Any updates on this ?

**Comment by Xenwar** [comment]
Not yet. But, will update this issue by tomorrow.

**Comment by Xenwar** [comment]
> > ok, then this is something that needs to be investigated.
> 
> I agree. It might better to split this issue into two, if current investigations do not yield fruit.
> 
>     1. Why does it fail to create the new node successfully.
> 
>     2. Why is the original node affected/changed before the success of the new one.

For the second case, please refer to https://github.com/metal3-io/cluster-api-provider-metal3/issues/74

**Comment by Xenwar** [comment]
> > ok, then this is something that needs to be investigated.
> 
> I agree. It might better to split this issue into two, if current investigations do not yield fruit.
> 
>     1. Why does it fail to create the new node successfully.
> 
>     2. Why is the original node affected/changed before the success of the new one.

@maelk @stbenjam 
I have created a new issue for second one. #74.
I am still able to reproduce the first one, but not consistently. I will take it as a side task to gather more evidence while working on other upgrade issues.

**Comment by Xenwar** [comment]
> > > ok, then this is something that needs to be investigated.
> > 
> > 
> > I agree. It might better to split this issue into two, if current investigations do not yield fruit.
> > ```
> > 1. Why does it fail to create the new node successfully.
> > 
> > 2. Why is the original node affected/changed before the success of the new one.
> > ```
> 
> @maelk @stbenjam
> I have created a new issue for second one. #74.
> I am still able to reproduce the first one, but not consistently. I will take it as a side task to gather more evidence while working on other upgrade issues.

The first case is out of scope and not a valid use case. Closing this issue.
